Bottom: 081d95eb5ac612b1f1136ba159a2ccb5716a75ee
Top:    05e8f43e60ef29b0be5ecb4cc8bc13ea5a348fa3
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-10-01 11:22:17 +0100

hid/gtk: (GL Renderer) Use direct rendering for off-screen pixmaps

We use direct rendering for the rest (where possible), so why slow down
pixmap rendering. Also fixes a bug on my pcb+gl_experimental branch
where pixel shaders can't be shared between the direct rendered widgets
and the pixmap rendering code.


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index b9a569f..069380e 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1107,7 +1107,7 @@ ghid_render_pixmap (int cx, int cy, double zoom, int width, int height, int dept
   pixmap = gdk_pixmap_new (NULL, width, height, depth);
   glpixmap = gdk_pixmap_set_gl_capability (pixmap, glconfig, NULL);
   gldrawable = GDK_GL_DRAWABLE (glpixmap);
-  glcontext = gdk_gl_context_new (gldrawable, NULL, FALSE, GDK_GL_RGBA_TYPE);
+  glcontext = gdk_gl_context_new (gldrawable, NULL, TRUE, GDK_GL_RGBA_TYPE);
 
   /* Setup zoom factor for drawing routines */
